script/validate-data/index.ts

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env npx ts-node
+import { promises as fs } from "fs";
+import { safeLoad } from "js-yaml";
+import { basename, extname, join } from "path";
+import { Validator as validator } from "jsonschema";
+import { endGroup, error, info, setFailed, startGroup } from '@actions/core';
+
+interface WorkflowWithErrors {
+  id: string;
+  errors: string[];
+}
+
+interface WorkflowProperties {
+  name: string;
+  description: string;
+  iconName: string;
+  categories: string[];
+}
+
+const propertiesSchema = {
+  type: "object",
+  properties: {
+    name: { type: "string", required: true },
+    description: { type: "string", required: true },
+    iconName: { type: "string", required: true },
+    categories: {
+      anyOf: [
+        {
+          type: "array",
+          items: { type: "string" }
+        },
+        {
+          type: "null",
+        }
+      ],
+      required: true
+    },
+  }
+}
+
+async function checkWorkflows(folders: string[]): Promise<WorkflowWithErrors[]> {
+  const result: WorkflowWithErrors[] = []
+
+  for (const folder of folders) {
+    const dir = await fs.readdir(folder, {
+      withFileTypes: true,
+    });
+
+    for (const e of dir) {
+      if (e.isFile()) {
+        const fileType = basename(e.name, extname(e.name))
+
+        const workflowFilePath = join(folder, e.name);
+        const propertiesFilePath = join(folder, "properties", `${fileType}.properties.json`)
+
+        const errors = await checkWorkflow(workflowFilePath, propertiesFilePath);
+        if (errors.errors.length > 0) {
+          result.push(errors)
+        }
+      }
+    }
+  }
+
+  return result;
+}
+
+async function checkWorkflow(workflowPath: string, propertiesPath: string): Promise<WorkflowWithErrors> {
+  let workflowErrors: WorkflowWithErrors = {
```

Comment:
```suggestion
  const workflowErrors: WorkflowWithErrors = {
```

Reply:
I find JS really weird with `const` being mutable, which is why I tend to use `let` in this case.  For me it shows the intent that it'll be something that changes (as we add errors into it). Fair?